packages/flutter/lib/src/widgets/gesture_detector.dart

```diff
@@ -335,7 +335,7 @@ class RawGestureDetectorState extends State<RawGestureDetector> {
   /// the gesture detector should be enabled.
   void replaceGestureRecognizers(Map<Type, GestureRecognizerFactory> gestures) {
     assert(() {
-      if (!RenderObject.debugDoingLayout) {
+      if (!Renderer.instance.pipelineOwner.debugDoingLayout) {
```

Comment:
This assert will trigger if you try this for a tree owned by a different owner.  I'd probably add a TODO comment to that effect so that when this triggers for someone in the future they'll have a breadcrumb to follow to understand how to resolve the issue.